release notes/v0.41.0.md

```diff
@@ -0,0 +1,64 @@
+- [#2705](https://github.com/grafana/k6/pull/2705) fixed a panic in the `k6/net/grpc` module ([#2661](https://github.com/grafana/k6/issues/2661)). Thanks, @c47gao and @robpickerill!
+- [#2738](https://github.com/grafana/k6/pull/2738) fixed a panic when a Promise was rejected with an `undefined` reason.
+- [#2739](https://github.com/grafana/k6/pull/2739) fixed hidden stack traces in certain types of errors originating from k6's Go code.
+
+## Maintenance and internal improvements
+
+We had a few minor changes in this release:
+
+- [#2687](https://github.com/grafana/k6/pull/2687) improved our logging tests. Thanks, @nsmith5!
+- [#2696](https://github.com/grafana/k6/pull/2696) updated the used Go version to 1.19.x and the Alpine version in our Docker image to 3.16. Thanks, @agilob!
+- [#2707](https://github.com/grafana/k6/pull/2707), [#2708](https://github.com/grafana/k6/pull/2708), [#2709](https://github.com/grafana/k6/pull/2709), [#2710](https://github.com/grafana/k6/pull/2710) updated most of the Go dependencies k6 has.
+- [#2716](https://github.com/grafana/k6/pull/2716) refactored how custom JS tags are applied to metrics and cleaned up validation for invalid tag values.
+
+As well as a couple of significant improvements that will allow us to develop exciting new features soon:
+
+### Metric time series ([#2594](https://github.com/grafana/k6/pull/2594))
+
+Previous to [#2594](https://github.com/grafana/k6/pull/2594), k6 didn't have an efficient way to group metric samples with the same tags. It meant that a whole class of applications for processing and aggregating metrics were nearly impossible to do or, at best, very inefficient.
+
+At the cost of some minor [breaking changes](#breaking-changes), we now have a performant internal representation to group metric samples with the same tags at the time of the action that generated them, i.e. the time of metric measurement. It allows us to efficiently group samples for the same action (e.g. an HTTP request to a specific URL) over time and construct time series with them.
+
+### Internal support for high-cardinality metrics metadata ([#2726](https://github.com/grafana/k6/pull/2726), [#2727](https://github.com/grafana/k6/pull/2727))
+
+Besides the previously described efficient grouping of metric samples into time series, which are meant for relatively low-cardinality data, k6 needed some way to attach high-cardinality metadata as well. This is necessary for data like unique or random identifiers, e.g. Trace and Span IDs (in distributed tracing), user IDs (e.g. in tests with huge data sets), etc.
+
+This has been added in k6 v0.41.0, and the `vu` and `iter` system tags have been transformed into metadata (see the [breaking changes section](#breaking-changes) above), but it is not yet accessible from user scripts. There is no JavaScript API to modify these metadata, only built-in k6 Go modules and [xk6](https://github.com/grafana/xk6) Go extensions can make use of it, for now.
```

Comment:
```suggestion
This has been added in k6 v0.41.0, and the `vu` and `iter` system tags have been transformed into metadata (see the [breaking changes section](#breaking-changes) above), but it is not yet accessible from user scripts. There is no JavaScript API to modify this metadata, only built-in k6 Go modules and [xk6](https://github.com/grafana/xk6) Go extensions can make use of it&mdash;for now.
```

Reply:
Maybe the dash can add a little moment of suspense.

Reply:
But it makes the text more unreadable in markdown, so I'd prefer a comma or a normal dash, or just using "—" directly.

```suggestion
This has been added in k6 v0.41.0, and the `vu` and `iter` system tags have been transformed into metadata (see the [breaking changes section](#breaking-changes) above), but it is not yet accessible from user scripts. There is no JavaScript API to modify this metadata, only built-in k6 Go modules and [xk6](https://github.com/grafana/xk6) Go extensions can make use of it, for now.
```